### DISCUSSION TITLE
How do i change the default keymaps?

### DISCUSSION THREAD

**Discussion by drvxksh** [zed-industries/zed#20770] [Feature Requests]
I know i can change the base map from Jetbrains to vs code etc, though i hardly notice any change when i change this option. Other than this i Open the default key maps and i can just see it. I am not able to make any changes into it. How can i modify these settings according to what i want?

**Comment by SomeoneToIgnore** [comment]
<img width="575" alt="image" src="https://github.com/user-attachments/assets/b4225f02-717f-465c-81da-b4d84958f9a1">

You open a keymap, and then place whatever custom items you need on top

<img width="1728" alt="image" src="https://github.com/user-attachments/assets/64401b4f-924e-4e40-8128-f7c2d820f386">

**Reply by drvxksh** [reply]
I have seen this. so is this the way to overwrite things? I thought there might be some way to make the desired changes in the default one only

**Reply by SomeoneToIgnore** [reply]
Defaults are not really changeable, but you can re-specify the same context + action with a `null` keybinding.
That will suppress this binding from being called, and anything new can be added into that local file instead.

IIRC there's an issue that not all of such things can be truly disabled, but at least there's a way to debug keybindings and contexts, as things may get tricky quite fast:

<img width="663" alt="image" src="https://github.com/user-attachments/assets/938df1e7-fe97-4f48-ab88-106c09d66885">
<img width="1728" alt="image" src="https://github.com/user-attachments/assets/79d16667-0f4f-4d33-b203-372b0bb4094f">

**Reply by drvxksh** [reply]
Makes sense. Thank you mate! really appreciate the help.

**Comment by SomeoneToIgnore** [comment]
See also https://zed.dev/docs/key-bindings#tips-and-tricks
